app/src/components/SectionPicker.tsx

```diff
@@ -78,13 +95,25 @@ export default function SectionPicker(props: Props) {
                         "bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg p-2.5 mb-0.5"
                     }
                     placeholder={getSearchByStringExample(searchByString)}
-                    value={props.searchText}
-                    onChange={(e) => props.setSearchText(e.target.value)}
+                    value={searchText}
+                    onChange={(e) => setSearchText(e.target.value)}
                     autoComplete={"off"}
+                    onKeyDown={(e) => {
+                        if (e.key === "Enter") {
+                            searchCourse();
+                        }
+                    }}
                 />
+
+                <button
+                    onClick={searchCourse}
+                    className="bg-sky-500 hover:bg-sky-400 border border-blue-300 text-white text-sm rounded-lg p-2.5 mr-1"
```

Comment:
Feature: We can make the button appear when the query text change!

Having the button disappear when querying and the cursor change tells the user that the app has accepted the request for information and is processing it.

Reply:
I don't think that's necessary because there is a "Loading..." indicator to show the user that the app has processed their query (we can improve it by making it more obvious). Making elements disappear from the UI seems like bad UI/UX

Reply:
Disappearing inputs are bad UX.

Maybe just put a small spinning circle to the right of the search button when it is fetching. I would still like to display cached results when possible (hence not putting the "Loading..." at the top).